Description: Use the system libpcre2
Author: Boian Bonev <bbonev@ipacct.com>
Origin: other
Forwarded: not-needed
Last-Update: 2021-11-23

--- a/vstring/makefile
+++ b/vstring/makefile
@@ -27,7 +27,7 @@ link: mm_update link-modules link-vstrin
 AR ?= ar
 LD = $(CXX)
 MKDIR = mkdir -p
-MODULES = pcre2
+MODULES =
 RANLIB ?= ranlib
 RMDIR = rm -rf
 RMFILE = rm -f
@@ -40,7 +40,7 @@ CC_1       = $(CXX)
 LD_1       = $(CXX)
 AR_1       = $(AR) rv
 RANLIB_1   = $(RANLIB)
-CCFLAGS_1  = -I. -Ipcre2 -O2 $(CFLAGS) $(CPPFLAGS) $(CCDEF)  
+CCFLAGS_1  = -I. -O2 $(CFLAGS) $(CPPFLAGS) $(CCDEF)  
 LDFLAGS_1  = $(LDFLAGS) $(LDDEF) 
 DEPFLAGS_1 = 
 ARFLAGS_1  = 
@@ -95,8 +95,8 @@ CC_2       = $(CXX)
 LD_2       = $(CXX)
 AR_2       = $(AR) rv
 RANLIB_2   = $(RANLIB)
-CCFLAGS_2  = -I. -Ipcre2 -O2 -g $(CFLAGS) $(CPPFLAGS) $(CCDEF)  
-LDFLAGS_2  = -Lpcre2 -lpcre2 $(LDFLAGS) $(LDDEF) 
+CCFLAGS_2  = -I. -O2 -g $(CFLAGS) $(CPPFLAGS) $(CCDEF)  
+LDFLAGS_2  = -lpcre2-8 $(LDFLAGS) $(LDDEF) 
 DEPFLAGS_2 = 
 ARFLAGS_2  = 
 TARGET_2   = test
@@ -149,16 +149,12 @@ link-test: .OBJ.test $(OBJ_2)
 ### MODULES ####################################################################
 
 modules:
-	$(MAKE) -C pcre2 
 
 clean-modules:
-	$(MAKE) -C pcre2 clean
 
 rebuild-modules:
-	$(MAKE) -C pcre2 rebuild
 
 link-modules:
-	$(MAKE) -C pcre2 link
 
 
 mm_update:
--- a/vfu/makefile
+++ b/vfu/makefile
@@ -39,8 +39,8 @@ CC_1       = $(CXX)
 LD_1       = $(LD)
 AR_1       = $(AR) rv
 RANLIB_1   = $(RANLIB)
-CCFLAGS_1  = -I../vstring -I../vstring/pcre2 -I../vslib -I/usr/include/ncurses -O2 $(CFLAGS) $(CPPFLAGS) $(CCDEF)  
-LDFLAGS_1  = -L../vstring -L../vstring/pcre2 -L../vslib -lvstring -lvslib -lvscon -lpcre2 -lncurses $(LDFLAGS) $(LDDEF) 
+CCFLAGS_1  = -I../vstring -I../vslib -I/usr/include/ncurses -O2 $(CFLAGS) $(CPPFLAGS) $(CCDEF)  
+LDFLAGS_1  = -L../vstring -L../vslib -lvstring -lvslib -lvscon -lpcre2-8 -lncurses $(LDFLAGS) $(LDDEF) 
 DEPFLAGS_1 = 
 ARFLAGS_1  = 
 TARGET_1   = vfu
@@ -132,8 +132,8 @@ CC_2       = $(CXX)
 LD_2       = $(LD)
 AR_2       = $(AR) rv
 RANLIB_2   = $(RANLIB)
-CCFLAGS_2  = -I../vstring -I../vstring/pcre2 -I../vslib -I../yascreen -DUSE_YASCREEN -O2 $(CFLAGS) $(CPPFLAGS) $(CCDEF)  
-LDFLAGS_2  = -L../vstring -L../vstring/pcre2 -L../vslib -L../yascreen -lvstring -lvslib -lvscony -lpcre2 ../yascreen/libyascreen.a -lrt $(LDFLAGS) $(LDDEF) 
+CCFLAGS_2  = -I../vstring -I../vslib -I../yascreen -DUSE_YASCREEN -O2 $(CFLAGS) $(CPPFLAGS) $(CCDEF)  
+LDFLAGS_2  = -L../vstring -L../vslib -L../yascreen -lvstring -lvslib -lvscony -lpcre2-8 ../yascreen/libyascreen.a -lrt $(LDFLAGS) $(LDDEF) 
 DEPFLAGS_2 = 
 ARFLAGS_2  = 
 TARGET_2   = vfu.yas
@@ -225,8 +225,8 @@ CC_3       = $(CXX)
 LD_3       = $(LD)
 AR_3       = $(AR) rv
 RANLIB_3   = $(RANLIB)
-CCFLAGS_3  = -I../vstring -I../vstring/pcre2 -I../vslib -I/usr/include/ncurses -O0 -g $(CFLAGS) $(CPPFLAGS) $(CCDEF)  
-LDFLAGS_3  = -L../vstring -L../vstring/pcre2 -L../vslib -lvstring -lvslib -lvscon -lpcre2 -lncurses -g $(LDFLAGS) $(LDDEF) 
+CCFLAGS_3  = -I../vstring -I../vslib -I/usr/include/ncurses -O0 -g $(CFLAGS) $(CPPFLAGS) $(CCDEF)  
+LDFLAGS_3  = -L../vstring -L../vslib -lvstring -lvslib -lvscon -lpcre2-8 -lncurses -g $(LDFLAGS) $(LDDEF) 
 DEPFLAGS_3 = 
 ARFLAGS_3  = 
 TARGET_3   = vfu-debug
@@ -318,8 +318,8 @@ CC_4       = $(CXX)
 LD_4       = $(LD)
 AR_4       = $(AR) rv
 RANLIB_4   = $(RANLIB)
-CCFLAGS_4  = -I../vstring -I../vstring/pcre2 -I../vslib -I../yascreen -DUSE_YASCREEN -O0 -g $(CFLAGS) $(CPPFLAGS) $(CCDEF)  
-LDFLAGS_4  = -L../vstring -L../vstring/pcre2 -L../vslib -L../yascreen -lvstring -lvslib -lvscony -lpcre2 ../yascreen/libyascreen.a -lrt -g $(LDFLAGS) $(LDDEF) 
+CCFLAGS_4  = -I../vstring -I../vslib -I../yascreen -DUSE_YASCREEN -O0 -g $(CFLAGS) $(CPPFLAGS) $(CCDEF)  
+LDFLAGS_4  = -L../vstring -L../vslib -L../yascreen -lvstring -lvslib -lvscony -lpcre2-8 ../yascreen/libyascreen.a -lrt -g $(LDFLAGS) $(LDDEF) 
 DEPFLAGS_4 = 
 ARFLAGS_4  = 
 TARGET_4   = vfu.yas-debug
--- a/vslib/makefile
+++ b/vslib/makefile
@@ -39,7 +39,7 @@ CC_1       = $(CXX)
 LD_1       = $(CXX)
 AR_1       = $(AR) rv
 RANLIB_1   = $(RANLIB)
-CCFLAGS_1  = -I../vstring -I../vstring/pcre2 -I. -O2 $(CFLAGS) $(CPPFLAGS) $(CCDEF)  
+CCFLAGS_1  = -I../vstring -I. -O2 $(CFLAGS) $(CPPFLAGS) $(CCDEF)  
 LDFLAGS_1  = $(LDFLAGS) $(LDDEF) 
 DEPFLAGS_1 = 
 ARFLAGS_1  = 
@@ -122,7 +122,7 @@ CC_2       = $(CXX)
 LD_2       = $(CXX)
 AR_2       = $(AR) rv
 RANLIB_2   = $(RANLIB)
-CCFLAGS_2  = -I../vstring -I../vstring/pcre2 -I. -I/usr/include/ncurses -O2 $(CFLAGS) $(CPPFLAGS) $(CCDEF)  
+CCFLAGS_2  = -I../vstring -I. -I/usr/include/ncurses -O2 $(CFLAGS) $(CPPFLAGS) $(CCDEF)  
 LDFLAGS_2  = $(LDFLAGS) $(LDDEF) 
 DEPFLAGS_2 = 
 ARFLAGS_2  = 
@@ -185,7 +185,7 @@ CC_3       = $(CXX)
 LD_3       = $(CXX)
 AR_3       = $(AR) rv
 RANLIB_3   = $(RANLIB)
-CCFLAGS_3  = -I../vstring -I../vstring/pcre2 -I. -I../yascreen -DUSE_YASCREEN -O2 $(CFLAGS) $(CPPFLAGS) $(CCDEF)  
+CCFLAGS_3  = -I../vstring -I. -I../yascreen -DUSE_YASCREEN -O2 $(CFLAGS) $(CPPFLAGS) $(CCDEF)  
 LDFLAGS_3  = $(LDFLAGS) $(LDDEF) 
 DEPFLAGS_3 = 
 ARFLAGS_3  = 
@@ -248,8 +248,8 @@ CC_4       = $(CXX)
 LD_4       = $(CXX)
 AR_4       = $(AR) rv
 RANLIB_4   = $(RANLIB)
-CCFLAGS_4  = -g -I../vstring -I../vstring/pcre2 -I. -I../yascreen -DUSE_YASCREEN -O0 -DTEST $(CFLAGS) $(CPPFLAGS) $(CCDEF)  
-LDFLAGS_4  = -g -L../vstring -L../vstring/pcre2 -L. -lvstring -lvslib -lvscon -lpcre2 -lncurses $(LDFLAGS) $(LDDEF) 
+CCFLAGS_4  = -g -I../vstring -I. -I../yascreen -DUSE_YASCREEN -O0 -DTEST $(CFLAGS) $(CPPFLAGS) $(CCDEF)  
+LDFLAGS_4  = -g -L../vstring -L. -lvstring -lvslib -lvscon -lpcre2-8 -lncurses $(LDFLAGS) $(LDDEF) 
 DEPFLAGS_4 = 
 ARFLAGS_4  = 
 TARGET_4   = test
